### ISSUE TITLE
Feature request: Ability to crop images, or do ROI analysis

### ISSUE THREAD

**Issue by tdbjacobs** [ContactEngineering/TopoBank#613]
Would it be possible to allow the user to crop out topographies? 

I could see this being useful if there is an obviously artifacted region, or if only a subset of the data is a region of interest. 

And/or we could also allow the user to perform analysis in only a specific Region Of Interest (ROI).

Any thoughts on how difficult this would be? 

-Tevis

**Comment by mcrot** [comment]
Well, the plotting library I use (bokeh) supports selection of a subset of data. I've never tried it, but probably it's possible
to send this selection choice (e.g. two points) back to the server. 
Then the user could select data and press a button which generates a new topography
in the same surface, based on the selected data and the same meta data. This would be an extra button on the detail view of a topography. This would also make sense for line scans, I guess.

**Comment by pastewka** [comment]
Makes sense... Do we want to create a completely new "measurement" or should be current measurement be altered?

**Comment by mcrot** [comment]
Maybe both can be useful: "selection as new" and "replace with selection". 

If we would just alter the measurement, we must delete the original data and 
you cannot have both, e.g. for comparison in the plots. 

If we would create a new one, the user could decide whether to delete the original measurement or not (=both options).

Another idea, but much more complex: The selection is used as an inverted "mask" and implemented like a filter. This could be 
then switched on and off.

**Comment by pastewka** [comment]
I would not delete the original but implement the cropping as a filter as you suggest.

**Comment by mcrot** [comment]
Yes, I would also not delete the original. The filter option this was my second idea. 
The first option I mentioned is to just generate a new topography as copy of the original with cropped data
with all meta data (also sharing?). Maybe plus providing a checkbox for deletion of the original data.

Let's also consider this in comparison to the filter option. What are the pros and cons?

Pros and cons of "Filter option", compared to just make a copy:
- pro: less data in the backend
- pro: more transparent what is happening with the data
- pro: you can go back to original data and make a new selection
- con: it's not so easy to compare cropped data with the original in one plot, because only the filtered data would be effective in analyses (however "copy" could be a button for all topographies, if useful)
- con: currently more difficult to implement, since filters are not there yet

**Comment by pastewka** [comment]
Would the copied topograpghy then become part of the same surface? In that case, one would need an option to exclude a topography from a surface since I believe you would only want the cropped one.

**Comment by tdbjacobs** [comment]
Yes, these are all good points and I'm not sure of the best approach.

I am concerned about making a copy because it may be tricky to keep track
of all the copies. A user may try a few different crops and it wouldn't be
a good situation if they wound up with a few different copies in the same
surface (assuming they forgot to delete).

Maybe the standard could be filters, but then there could be an option (not
default) to "Create new topography from crop"? Would that be possible?

-Tevis



Tevis Jacobs
William Kepler Whiteford Faculty Fellow and Associate Professor
Mechanical Engineering and Materials Science
University of Pittsburgh
http://www.engineering.pitt.edu/JacobsLab/
Mail: 636 Benedum Hall, 3700 O'Hara St., Pittsburgh, PA 15261
Office: 538-E Benedum Hall, 412-624-9736
***@***.***

**Comment by pastewka** [comment]
We need to implement filters anyway, and I think this is a classical example of those filters. With regards to comparing cropped and non-cropped data, we could create and option to explicitly deactivate filters when displaying the PSD, ACF, etc. I don't think this should be a priority though.

I am adding this to version 1.1. which should be the one that implements filters.

**Comment by tdbjacobs** [comment]
Hey All, did we ever move forward on the cropping/ROI capability? I know we have a lot of different things we are implementing!

**Comment by pastewka** [comment]
This is something post v1.0, so it's on the near-term list.
